packages/discord.js/src/util/Invites.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const { InviteType } = require('discord-api-types/v10');
+const { BaseInvite } = require('../structures/BaseInvite.js');
+const { GroupDMInvite } = require('../structures/GroupDMInvite.js');
+const { GuildInvite } = require('../structures/GuildInvite.js');
+
+/**
+ * Any invite.
+ * @typedef {GuildInvite|GroupDMInvite} Invite
+ */
+
+const InviteTypeToClass = {
+  [InviteType.Guild]: GuildInvite,
+  [InviteType.GroupDM]: GroupDMInvite,
+};
+
+/**
+ * Creates an invite.
+ * @param {Client} client The client
+ * @param {Object} data The data
+ * @returns {BaseInvite}
+ * @ignore
+ */
+function createInvite(client, data) {
```

Comment:
This means it could happen to have an actual instance of BaseInvite, not a child class. But BaseInvite is documented to be abstract.